app/src/main/java/xyz/whereuat/whereuat/ui/views/AddContactDialogFragment.java

```diff
@@ -0,0 +1,119 @@
+package xyz.whereuat.whereuat.ui.views;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import xyz.whereuat.whereuat.AsyncExecutor;
+import xyz.whereuat.whereuat.Constants;
+import xyz.whereuat.whereuat.MainActivity;
+import xyz.whereuat.whereuat.PendingRequestsActivity;
+import xyz.whereuat.whereuat.R;
+import xyz.whereuat.whereuat.utils.ContactUtils;
+import xyz.whereuat.whereuat.utils.PendingRequestUtils;
+
+/**
+ * This class is used for handling the dialog that is displayed when the user tries to add pending
+ * request for a number that was not found in the phonebook.
+ */
+public class AddContactDialogFragment extends DialogFragment {
```

Comment:
A little confused by what you mean here, however I do like the term "pending contact". I keep finding "pending request" to be rather confusing because the requests don't actually pend, they just add the contact to holding area until you approve them.